src/rest-api/content/add.ts

```diff
@@ -0,0 +1,49 @@
+import { z } from 'zod';
+import { ApiRequestParams, handleRequest } from '../../API';
+import {
+  PloneClientConfig,
+  PloneClientConfigSchema,
+} from '../../interfaces/config';
+import { AddContentResponse } from '../../interfaces/content/add';
+
+export const createContentDataSchema = z.object({
+  '@id': z.string().optional(),
+  '@type': z.string().optional(),
+  title: z.string(),
+});
+
+export const createContentArgsSchema = z.object({
+  path: z.string(),
+  data: createContentDataSchema,
+  config: PloneClientConfigSchema,
+});
+
+export type CreateContentArgs = z.infer<typeof createContentArgsSchema>;
```

Comment:
Args types are defined as zod objects and Typescript types are inferred from that wherever needed to avoid duplication.